sktime/transformations/series/window_summarizer.py

```diff
@@ -0,0 +1,360 @@
+        "fit-in-transform": False,  # is fit empty and can be skipped? Yes = True
+        "transform-returns-same-time-index": False,
+        # does transform return have the same time index as input X
+    }
+
+    def __init__(
+        self,
+        summarizer=None,
+        n_jobs=-1,
+        target_cols=None,
+    ):
+
+        # self._converter_store_X = dict()
+        self.summarizer = summarizer
+        self.n_jobs = n_jobs
+        self.target_cols = target_cols
+
+        super(WindowSummarizer, self).__init__()
+
+    # Get extraction parameters
+    def _fit(self, X, y=None):
+        """Fit transformer to X and y.
+
+        Private _fit containing the core logic, called from fit
+
+        Parameters
+        ----------
+        X : pd.Series or pd.DataFrame
+        y : None
+
+        Returns
+        -------
+        X: pd.DataFrame
+        self: reference to self
+        """
+        X_name = get_name_list(X)
+
+        if X_name is None:
+            self._X_rename = "var_0"
+            X_name = ["var_0"]
+        else:
+            self._X_rename = None
+
+        if self.target_cols is None:
+            self._target_cols = [X_name[0]]
+        else:
+            self._target_cols = self.target_cols
+
+        if self.target_cols is not None:
+            if not all(x in X_name for x in self.target_cols):
+                missing_cols = [x for x in self.target_cols if x not in X_name]
+                raise ValueError(
+                    "target_cols "
+                    + " ".join(missing_cols)
+                    + " specified that do neither exist in X (or y resp.)"
+                )
+
+        if self.summarizer is None:
+            func_dict = pd.DataFrame(
+                {
+                    "lag": ["lag", [[1, 0]]],
+                }
+            ).T.reset_index()
+        else:
+            func_dict = pd.DataFrame(self.summarizer).T.reset_index()
+
+        func_dict.rename(
+            columns={"index": "name", 0: "win_summarizer", 1: "window"},
+            inplace=True,
+        )
+        func_dict = func_dict.explode("window")
+        self._func_dict = func_dict
+        self._truncate_start = func_dict["window"].apply(lambda x: x[0] + x[1]).max()
+
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        Parameters
+        ----------
+        X : Series or Panel
+        y : Series or Panel
+
+        Returns
+        -------
+        transformed version of X
+        """
+        # input checks
+        # if not isinstance(y, pd.DataFrame):
+        #     y = pd.DataFrame(y)
+
+        # if not isinstance(X, pd.DataFrame):
+        #     X = pd.DataFrame(X)
+
+        # if self._y_rename is not None:
+        #     y.columns = [self._y_rename]
+        X.columns = X.columns.map(str)
+
+        if self._X_rename is not None:
+            X.columns = [self._X_rename]
```

Comment:
this is a side effect on `X`? Let's not rename the columns of `X`.

Reply:
Actually this is a left over from a previous approach with no impact. Removed it.